docs/docs/Integrations/Google/integrations-setup-google-cloud-vertex-ai-langflow.md

```diff
@@ -0,0 +1,44 @@
+---
+title: 'Integrate Google Cloud Vertex AI with Langflow'
+slug: /integrations-setup-google-cloud-vertex-ai-langflow
+sidebar_position: 2
+description: "A comprehensive guide on creating a Google OAuth app, obtaining tokens, and integrating them with Langflow's Google components."
+---
+
+Langflow integrates with the [Google Vertex AI API](https://console.cloud.google.com/marketplace/product/google/aiplatform.googleapis.com) for authenticating the [Vertex AI embeddings model](/components-embedding-models#vertexai-embeddings) and [Vertex AI](/components-models#vertexai) components.
+
+Learn how to create a service account JSON in Google Cloud to authenticate Langflow’s Vertex AI components.
+
+## Create a project and enable the Vertex AI API
+
+1. Navigate to the [Google Cloud Console](https://console.cloud.google.com/).
+2. Click **Select a project**, and then click **New Project** to create a new project.
+3. To enable the Vertex AI API, navigate to the [Vertex AI API](https://console.cloud.google.com/marketplace/product/google/aiplatform.googleapis.com).
+4. Select your project, and then click **Enable**.
+
+## Create a service account
+
+1. Navigate to **IAM & Admin**, and then click **Service Accounts**.
+2. To create a new service account, click **Create Service Account**.
+3. Provide a name and a description, and then click on **Create and Continue**.
+4. Assign the **Vertex AI Service Agent** role to your new account.
+This role allows Langflow to access Vertex AI resources.
+For more information, see [Vertex AI access control with IAM](https://cloud.google.com/vertex-ai/docs/general/access-control).
+5. To generate a new JSON key for the service account, navigate to your service account.
+6. Click **Add Key**, and then click **Create new key**.
+7. Under **Key type**, select **JSON**, and then click **Create**.
```

Comment:
This is still documenting the steps for a 3rd-party application. Provide generic instructions with links to the appropriate docs for the Google Cloud Console.